lib/ansible/utils/collection_loader.py

```diff
@@ -47,6 +49,15 @@ def __init__(self, config=None):
         elif self._n_configured_paths is None:
             self._n_configured_paths = []
 
+        # Append all ``ansible_collections`` dirs from sys.path to the end
+        for path in sys.path:
+            sys_installed_path = os.path.join(os.path.abspath(path), 'ansible_collections')
+            if (
+                    os.path.isdir(sys_installed_path)
+                    and path not in self._n_configured_paths
+            ):
```

Comment:
This can be simplified and optimized:

1. Check `self._n_configured_paths` before calling `os.path.isdir` to avoid unnecessary filesystem checks.
2. Simplify the conditional to remove the intermediate variable since it is used only once.
3. Avoid calling `os.path.abspath` since it `os.path.isdir` will handle relative paths for us.

```suggestion
            if path not in self._n_configured_paths and os.path.isdir(os.path.join(path, 'ansible_collections')):
```